src/coreclr/vm/assemblyspec.cpp

```diff
@@ -564,7 +564,7 @@ void AssemblySpec::AssemblyNameInit(ASSEMBLYNAMEREF* pAsmName, PEImage* pImageIn
     {
         DWORD dwMachine, dwKind;
 
-        pImageInfo->GetPEKindAndMachine(&dwMachine,&dwKind);
+        pImageInfo->GetPEKindAndMachine(&dwKind, &dwMachine);
```

Comment:
The names of the local variables are swapped, but they're also swapped below when used to call the `AssemblyName.SetProcArchIndex`. So in the end the right values are passed in. It would be great if you could simply swap the variables in both places so that the names match the meaning.

This also means that there's no test for this (it should have failed with your change). It would be great if you could add a test for this. Probably somewhere around here: https://github.com/dotnet/runtime/tree/main/src/tests/Loader/binding/assemblies/basicapi/assemblynamector

Reply:
Thanks. Great to know that this isn't an actual bug. I am blocked by https://github.com/dotnet/runtime/issues/50175 for a test. Once I have that sorted out, I will update this PR.